backends/github/labels/github-labels.js

```diff
@@ -0,0 +1,165 @@
+		let toUpdate = data.filter(label => label.new_name || this.data.find(l => l.name === label.name && ["color", "default", "description"].some(prop => l[prop] !== label[prop])));
+
+		// Create new labels
+		// https://docs.github.com/en/rest/issues/labels?apiVersion=2022-11-28#create-a-label
+		if (toCreate.length) {
+			ret.created = await this.#write("create", toCreate, ref);
+		}
+
+		// Update existing labels
+		// https://docs.github.com/en/rest/issues/labels?apiVersion=2022-11-28#update-a-label
+		if (toUpdate.length) {
+			ret.updated = await this.#write("update", toUpdate, ref);
+		}
+
+		// Delete existing labels
+		// https://docs.github.com/en/rest/issues/labels?apiVersion=2022-11-28#delete-a-label
+		if (toDelete.length) {
+			ret.deleted = await this.#write("delete", toDelete, ref);
+		}
+
+		return ret;
+	}
+
+	/**
+	 * Delete all existing labels in one go.
+	 * @param {Object} ref
+	 * @returns {Promise<Object>} Promise that is resolved with an object based on the results of label deletion.
+	 */
+	async delete (ref = this.ref) {
+		return this.put([], {ref, force: true});
+	}
+
+	async #write (type, labels, ref) {
+		let method;
+		switch (type) {
+			case "update":
+				method = "PATCH";
+				break;
+
+			case "delete":
+				method = "DELETE";
+				break;
+
+			default:
+				method = "POST";
+		}
+
+		let result = await Promise.allSettled(labels.map(label => {
+			let apiCall = `${ref.apiCall}/${label.name}`, data = label, req;
+
+			switch (type) {
+				case "create":
+					apiCall = ref.apiCall;
+					break;
+
+				case "update":
+					if (label.new_name) {
+						data = {...label, name: label.new_name};
+					}
+					break;
+
+				case "delete":
+					data = null;
+					req = {responseType: "text"};
+					break;
+			}
+
+			return this.request(apiCall, data, method, req);
+		}));
+
+		let success = [], failure = [];
+		for (let [index, promise] of result.entries()) {
+			let label = labels[index];
+
+			if (promise.status === "fulfilled") {
+				if (type === "delete") {
+					// For deleted labels there is no additional information returned
+					label = { name: label.name };
+				}
+				else {
+					label = promise.value;
+				}
+
+				success.push(label);
+			}
+			else {
+				failure.push({label, reason: promise.reason});
+			}
+		}
+
+		if (success.length) {
+			console.info(`${this.constructor.phrase("success", type)} ${success.map(l => l.name).join(", ")}.`);
+		}
+
+		if (failure.length) {
+			console.warn(`${this.constructor.phrase("failure", type)} ${failure.map(l => l.name).join(", ")}.`);
+		}
+
+		return { success, failure };
+	}
+
+	// Why do we need this if we can provide static urls = []?
+	// All URLs supported by this backend are also supported by the parent (GithubAPI) class,
+	// and we want them to be parsed by the parent class, not this one.
+	// OTOH, we need a way to tell Madata when to use this class instead of the parent one.
+	static test (url) {
```

Comment:
I don't understand why this is needed. If it's something that won't be needed once we implement the ordering, then I wouldn't include it — the backend can be useful even if you have to specify it explicitly.

Reply:
Get rid of this part after refactoring. Thanks